src/dodal/utils.py

```diff
@@ -410,7 +412,27 @@ def is_any_device_factory(func: Callable) -> TypeGuard[AnyDeviceFactory]:
 
 
 def is_v2_device_type(obj: type[Any]) -> bool:
```

Comment:
Should: Would be nice to have a test that explicitly checks this works for generics.
Could: Also testing the edge cases of not passing a class etc.